### PR TITLE
Convert FBV into CBV

### LINKED ISSUE #33: Convert FBV classes to CBV classes

### PR DESCRIPTION
**Pull request by seymaozukanar**
All views are converted into CBV.

### CODE REVIEW

File: hipogram/posts/views.py [hunk 1]
```diff
@@ -38,63 +38,37 @@ def get_context_data(self):
...
-        else:
-            form = PostForm(instance=post)
 
-        return render(request, "update.html", {'form': form, 'post': post})
-    else:
-        messages.error(request, 'You are not the owner of this post, please log in.')
-        return redirect("users:login")
+class PostUpdateView(UpdateView):
+    model = Post
+    fields = ['text', 'tags']
```
Comment (KaratasFurkan) [inline hipogram/posts/views.py]: Being consistent about style is better IMO, so use `'` or `"` everywhere.

File: hipogram/posts/views.py [hunk 2]
```diff
@@ -38,63 +38,37 @@ def get_context_data(self):
...
-            return redirect("posts:list")
+            return self.form_valid(form)
         else:
-            form = PostForm(instance=post)
-
-        return render(request, "delete.html", {'form': form})
-    else:
-        messages.error(request, 'You are not the owner of this post, please log in.')
-        return redirect("users:login")
+            return self.form_invalid(form)
```
Comment (KaratasFurkan) [inline hipogram/posts/views.py]: Does this work? It seems it's not okay.

File: hipogram/posts/urls.py [hunk 3]
```diff
@@ -1,15 +1,14 @@
...
 app_name = "posts"
 
 urlpatterns = [
     path("", PostListView.as_view(), name="list"),
-    path('share/', views.post_new, name='post_new'),
-    path('update/<int:post_id>/', views.update_post, name='update_post'),
-    path('delete/<int:post_id>/', views.delete_post, name='delete_post'),
+    path('share/', PostCreateView.as_view(), name='share'),
+    path('delete/<int:post_id>/', PostDeleteView.as_view(), name='delete'),
+    path('update/<int:post_id>/', PostUpdateView.as_view(), name='update'),
```
Comment (KaratasFurkan) [inline hipogram/posts/urls.py]: New names are better :+1:

**Review by KaratasFurkan — CHANGES_REQUESTED** [review]

File: hipogram/posts/views.py (hunk 2 above)
Comment (seymaozukanar) [reply hipogram/posts/views.py]: It works, but it could be better.

Comment (KaratasFurkan) [reply hipogram/posts/views.py]: Alright, you can continue with other views (like & rate), success / fail messages, login / owner required implementations. We can try to make this better in the meeting.

**Comment by KaratasFurkan** [comment]
Wow, last commit is huge :slightly_smiling_face: I wasn't expecting that.

I didn't want to bother you, just wanted to give advice for future but there is no harm to clean whole code base at once of course.

File: hipogram/posts/views.py [hunk 4]
```diff
@@ -71,27 +71,27 @@ class PostUpdateView(UpdateView):
...
+class RateView(LoginRequiredMixin, View):
 
-    # if not form.is_valid():
-        # do something
+    def get(self, request, *args, **kwargs):
 
-        rate, _ = Rate.objects.get_or_create(user=request.user, post_id=post_id)
-        rate.value = form.data['value']
-        rate.save()
+        if request.method == 'POST':
```
Comment (KaratasFurkan) [inline hipogram/posts/views.py]: You need to override `post` method if you want to handle `POST` requests.

Comment (seymaozukanar) [reply hipogram/posts/views.py]: Makes sense, working on it.

File: hipogram/posts/views.py [hunk 5]
```diff
@@ -1,14 +1,24 @@
...
+
+
+class OwnerRequiredMixin(LoginRequiredMixin):
+
+    def dispatch(self, request, *args, **kwargs):
+        response = super().dispatch(request, *args, **kwargs)
+        post = self.get_object()
+        if post.created_by != self.request.user:
+            return self.handle_no_permission()
+        return response
```
Comment (KaratasFurkan) [inline hipogram/posts/views.py]: Looks nice :ok_hand: . We can try a little more to make it even more cleaner together in the meeting.

File: hipogram/posts/views.py [hunk 6]
```diff
@@ -31,104 +41,68 @@ def get_context_data(self):
...
+    pk_url_kwarg = "post_id"
 
-def update_post(request, post_id):
-    post = get_object_or_404(Post, id=post_id)
+    def dispatch(self, request, *args, **kwargs):
+        response = super().dispatch(request, *args, **kwargs)
+        post = self.get_object()
+        if post.created_by != self.request.user:
+            return self.handle_no_permission()
+        return response
```
Comment (KaratasFurkan) [inline hipogram/posts/views.py]: I believe you forgot to delete this.

**Review by KaratasFurkan — CHANGES_REQUESTED** [review]

File: hipogram/posts/views.py (hunk 5 above)
Comment (KaratasFurkan) [reply hipogram/posts/views.py]: I think it would be better on `posts/mixins.py`

Comment (seymaozukanar) [reply hipogram/posts/views.py]: I created one but now it doesn't work. I believe I have forgotten about something. Can you check?

**Review by KaratasFurkan — APPROVED** [review]

**Pull request merged**